bundles/CoreBundle/DependencyInjection/Compiler/VersionStorageAdapterPass.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Pimcore\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use Elasticsearch\Endpoints\Delete;
+use Pimcore\Model\Version\Adapter\DelegateVersionStorageAdapter;
+use Pimcore\Model\Version\Adapter\VersionStorageAdapterInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class VersionStorageAdapterPass implements CompilerPassInterface
+{
+    const TAG_NAME = 'pimcore.version.storage.adapter';
+
+    /**
+     * @inheritDoc
+     */
+    public function process(ContainerBuilder $container)
```

Comment:
We don't need that compiler pass at all, since the actual adapters have anyway `getStorageType()` implemented, which then can be called by the `DelegateVersionStorageAdapter`.